### DISCUSSION TITLE
Installing VMware on WSL2 for nested virtualization.

### DISCUSSION THREAD

**Discussion by Jymitar** [microsoft/WSL#11320] [General]
Hey guys,
Don't ask me why but I want to install vmware(or vbox) on WSL2 Ubuntu 22.04 running on Win11 Pro.
The WSL2 works ok and I have nested virtualization enabled
The problem is that when I try to install either of the hypervisors I get error message becaue of missing kernel headers. 
I have tried to build a custom image or install the headers directly from the repository to no avail.

Has anyone encountered such issue or managed to run vmware/vbox or any other app that requires headers to build the required modules? 

King Regards!

**Comment by gc-lg** [comment]
Hi ! Have you come up with a solution ? I'd like to do the same thing, but i'm unable to get vmware running inside wsl2...
